cmd/infra/powervs/destroy.go

```diff
@@ -0,0 +1,562 @@
+	"fmt"
+	"github.com/IBM-Cloud/power-go-client/power/models"
+	"io/ioutil"
+	"k8s.io/apimachinery/pkg/util/errors"
+	"strings"
+	"time"
+
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/IBM-Cloud/power-go-client/clients/instance"
+	"github.com/IBM-Cloud/power-go-client/ibmpisession"
+	"github.com/IBM/platform-services-go-sdk/resourcecontrollerv2"
+	"github.com/IBM/vpc-go-sdk/vpcv1"
+)
+
+const (
+	// Time duration for monitoring the resource readiness
+	cloudInstanceDeletionTimeout   = time.Minute * 10
+	powerVSResourceDeletionTimeout = time.Minute * 5
+	vpcResourceDeletionTimeout     = time.Minute * 2
+	dhcpServerDeletionTimeout      = time.Minute * 10
+
+	// Resource desired states
+	powerVSCloudInstanceRemovedState = "removed"
+	powerVSJobCompletedState         = "completed"
+	powerVSJobFailedState            = "failed"
+	dhcpInstanceShutOffState         = "SHUTOFF"
+)
+
+// DestroyInfraOptions command line options to destroy infra created in IBMCloud for Hypershift
+type DestroyInfraOptions struct {
+	InfraID                string
+	InfrastructureJson     string
+	ResourceGroup          string
+	PowerVSRegion          string
+	PowerVSZone            string
+	PowerVSCloudInstanceID string
+	PowerVSDhcpID          string
+	PowerVSCloudConnection string
+	VpcRegion              string
+	Vpc                    string
+	Debug                  bool
+}
+
+func NewDestroyCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:          "powervs",
+		Short:        "Destroys PowerVS infrastructure resources for a cluster",
+		SilenceUsage: true,
+	}
+
+	opts := DestroyInfraOptions{}
+
+	cmd.Flags().StringVar(&opts.InfraID, "infra-id", opts.InfraID, "Cluster ID with which to tag IBM Cloud resources")
+	cmd.Flags().StringVar(&opts.InfrastructureJson, "infrastructure-json", opts.InfrastructureJson, "Result of ./hypershift infra create powervs")
+	cmd.Flags().StringVar(&opts.ResourceGroup, "resource-group", opts.ResourceGroup, "IBM Cloud Resource Group")
+	cmd.Flags().StringVar(&opts.VpcRegion, "vpc-region", opts.VpcRegion, "IBM Cloud VPC Infra Region")
+	cmd.Flags().StringVar(&opts.Vpc, "vpc", opts.Vpc, "IBM Cloud VPC")
+	cmd.Flags().StringVar(&opts.PowerVSRegion, "powervs-region", opts.PowerVSRegion, "PowerVS Region")
+	cmd.Flags().StringVar(&opts.PowerVSZone, "powervs-zone", opts.PowerVSZone, "IBM Cloud PowerVS Zone")
+	cmd.Flags().StringVar(&opts.PowerVSCloudConnection, "powervs-cloud-connection", opts.PowerVSCloudConnection, "IBM Cloud PowerVS Cloud Connection")
+	cmd.Flags().StringVar(&opts.PowerVSCloudInstanceID, "powervs-cloud-instance-id", opts.PowerVSCloudInstanceID, "IBM PowerVS Cloud Instance ID")
+	cmd.Flags().BoolVar(&opts.Debug, "debug", opts.Debug, "Enabling this will result in debug logs will be printed")
+
+	cmd.MarkFlagRequired("resource-group")
+	cmd.MarkFlagRequired("infra-id")
+	cmd.MarkFlagRequired("powervs-region")
+	cmd.MarkFlagRequired("powervs-zone")
+	cmd.MarkFlagRequired("vpc-region")
```

Comment:
can the vpc region be different than the powervs-region?

Reply:
Yes it can